app/models/manageiq/providers/ibm_cloud/inventory/parser/power_virtual_servers.rb

```diff
@@ -107,6 +107,51 @@ def pvm_instances
         :value        => instance.proc_type,
         :read_only    => true
       )
+
+      # saving pin_policy
+      persister.vms_and_templates_advanced_settings.build(
+        :resource     => ps_vmi,
+        :name         => 'pin_policy',
+        :display_name => _('Pin Policy'),
+        :description  => _('VM Pin Policy'),
+        :value        => instance.pin_policy,
+        :read_only    => true
+      )
+
+      next unless instance.software_licenses.nil?
```

Comment:
I don't really like this because if someone adds a new section to the bottom of the method it won't get executed as this bails out from the entire method.  This method is huge, so it's possible someone might not see this bail out in the middle.

IMO this string building should be extracted to a method, then wrap the if statement around just the single persister...something like:

```ruby
    if instance.software_licenses
      license_description = build_license_description(instance)
      if license_description.present?
        persister.vms_and_templates_advanced_settings.build(
          :resource     => ps_vmi,
          :name         => 'software_licenses',
          :display_name => _('Software Licenses'),
          :description  => _('Software Licenses'),
          :value        => ldesc,
          :read_only    => true
        )
      end
    end

# Then way down below ...

private def build_license_description(instance)
  ldesc = ""

  if instance.software_licenses.ibmi_css
    ldesc = "IBMi Cloud Storage Solution (ibmiCSS), "
  end

  #...
end

```

Reply:
Thanks. I will make the changes.